addons/hr/models/hr_employee_public.py

```diff
@@ -81,14 +167,34 @@ def _compute_employee_id(self):
         for employee in self:
             employee.employee_id = self.env['hr.employee'].browse(employee.id)
 
+    def _compute_newly_hired(self):
+        self._compute_from_employee('newly_hired')
+
+    def _search_newly_hired(self, operator, value):
+        if operator not in ('in', 'not in'):
+            return NotImplemented
+        new_hire_field = self.env['hr.employee']._get_new_hire_field()
+        new_hires = self.env['hr.employee'].sudo().search([
+            (new_hire_field, '>', fields.Datetime.now() - timedelta(days=90))
+        ])
+        return [('id', operator, new_hires.ids)]
+
     @api.model
     def _get_fields(self):
-        return ','.join('emp.%s' % name for name, field in self._fields.items() if field.store and field.type not in ['many2many', 'one2many'])
+        return 'e.id AS id,e.name AS name,' + ','.join(
+            ('v.%s' if name in self.env['hr.version']._fields and self.env['hr.version']._fields[name].store else 'e.%s') % name
+            for name, field in self._fields.items()
+            if field.store and field.type not in ['many2many', 'one2many'] and name not in ['id', 'name'])
 
     def init(self):
         tools.drop_view_if_exists(self.env.cr, self._table)
         self.env.cr.execute("""CREATE or REPLACE VIEW %s as (
             SELECT
                 %s
-            FROM hr_employee emp
+            FROM hr_employee e
+            LEFT JOIN (
+                SELECT DISTINCT ON (employee_id) *
+                FROM hr_version
+                ORDER BY employee_id, date_version DESC
+            ) v ON v.employee_id = e.id
```

Comment:
There is no unique constraint on `(employee_id, date_version)`, therefore the `hr_version` corresponding to a given `hr_employee` may be nondeterministic.  You can avoid the nondeterminism by adding `id DESC` in the `ORDER BY` clause, or by adding the unique constraint which may prevent other kinds of mistakes...

Reply:
Indeed, you are totally right, thanks! 💪🏼 

We added a constraint as you suggested. 🙂